runtime/lua/vim/_defaults.lua

```diff
@@ -805,31 +805,50 @@ do
       -- an OSC 11 response from the terminal emulator. If the user has set
       -- 'background' explicitly then we will delete this autocommand,
       -- effectively disabling automatic background setting.
-      local force = false
+      local bg_response_received = false
+      local bg_detection_complete = false
       local id = vim.api.nvim_create_autocmd('TermResponse', {
         group = group,
         nested = true,
         desc = "Update the value of 'background' automatically based on the terminal emulator's background color",
         callback = function(args)
           local resp = args.data.sequence ---@type string
+
+          -- DA1 response that should come after the OSC 11 response if the
+          -- terminal supports it.
+          if string.match(resp, '^\x1b%[%?.-c$') then
+            bg_detection_complete = true
+            if not bg_response_received then
+              return true
+            end
+            return false
+          end
+
```

Comment:
We intentionally don't delete this autocommand today because we want to handle other OSC 11 responses that are triggered from theme change notifications. This PR would regress that feature (I guess we don't have a test for that or else the test should have failed).

**EDIT:** Ah I see, but if we receive the DA1 response without ever having received an OSC 11 response then we know the terminal doesn't support it, in which case removing the autocommand is ok. So this is fine actually.